homeassistant/components/s3backup/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""Integration for S3 Cloud Storage."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import boto3
+from botocore.exceptions import ClientError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import (
+    CONF_ACCESS_KEY,
+    CONF_BUCKET,
+    CONF_S3_URL,
+    CONF_SECRET_KEY,
+    DATA_BACKUP_AGENT_LISTENERS,
+    LOGGER,
+)
+
+type S3ConfigEntry = ConfigEntry["S3ConfigEntryData"]
+
+
+@dataclass(kw_only=True)
+class S3ConfigEntryData:
+    """Dataclass holding all config entry data for a S3 entry."""
+
+    bucket: str
+    client: boto3.session.Session.client
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: S3ConfigEntry) -> bool:
+    """Set up S3 from a config entry."""
+    LOGGER.info("Entry details: %s", entry)
+    # Create a session using your credentials
+    session = boto3.Session(
+        aws_access_key_id=entry.data[CONF_ACCESS_KEY],
+        aws_secret_access_key=entry.data[CONF_SECRET_KEY],
+    )
+
+    try:
+        # Create an S3 client
+        client = await hass.async_add_executor_job(
+            session.client,
+            "s3",
+            None,
+            None,
+            True,
+            None,
+            entry.data[CONF_S3_URL],
+        )
+        entry.runtime_data = S3ConfigEntryData(
+            client=client, bucket=entry.data[CONF_BUCKET]
+        )
+    except ClientError as err:
+        LOGGER.error("Error: %s", err)
+        raise ConfigEntryAuthFailed(
+            f"Invalid authentication token for S3 account: {err}"
+        ) from err
+    except Exception as err:  # noqa: BLE001
```

Comment:
Don't catch all exceptions. Always limit it to errors you expect it to be raised

Reply:
HI Paulus! Thank you for reviewing and taking your time. I am new in homeassistant and perhaps would need a little more help. But I am eager to learn. This is the list of exceptions botocore could raise:
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#botocore-exceptions
I can not see in the documentation of session.client which exceptions it could raise. I browsed through the code and I think that could be a lot. Many should be covered by ClientError. But how should I take care for the Exceptions I am not aware of? I thought the "final" exception is taking care for this. If I would remove it - and an uncatched exceptions is being raised - what happens then? What is your recommendation?